### PR TITLE
[Move Expando:circluar dependency in local service]Lps 191596 8 2

### PR DESCRIPTION
**Pull request by slnn**
https://liferay.atlassian.net/browse/LPS-191596

We need to break these circular dependencies since OSGi Declarative Services doesn’t accommodate circular references.

### CODE REVIEW

**Comment by catarineaguiar** [comment]
no regressions found

File: portal-impl/src/com/liferay/portlet/expando/service/impl/ExpandoValueLocalServiceImpl.java [hunk 1]
```diff
@@ -463,8 +463,12 @@ public ExpandoValue addValue(
 			String columnName, long classPK, Object data)
 		throws PortalException {
 
-		ExpandoColumn column = _expandoColumnLocalService.getColumn(
-			companyId, className, tableName, columnName);
+		ExpandoTable table = _expandoTablePersistence.fetchByC_C_N(
+			companyId, _classNameLocalService.getClassNameId(className),
+			tableName);
+
+		ExpandoColumn column = _expandoColumnPersistence.fetchByT_N(
```
Comment (shuyangzhou) [inline portal-impl/src/com/liferay/portlet/expando/service/impl/ExpandoValueLocalServiceImpl.java:485]: @slnn this is not an equivalent change. "expandoColumnLocalService.getColumn()" has a null check for the ExpandoTable, but here you use the table to call "table.getTableId()" which could lead to NPE.

I did not check all your changes, please make sure you take care all similar cases.

**Comment by shuyangzhou** [comment]
@slnn see comment above, please fix and resend.

File: portal-impl/src/com/liferay/portlet/expando/service/impl/ExpandoValueLocalServiceImpl.java (hunk 1 above)
Comment (slnn) [reply portal-impl/src/com/liferay/portlet/expando/service/impl/ExpandoValueLocalServiceImpl.java:485]: OK, got it

**Pull request closed without merging**